docs/3.0.0-beta.x/migration-guide/migration-guide-alpha.26-to-beta.md

```diff
@@ -0,0 +1,663 @@
+  customAction: async ctx => {
+    ctx.send({ ok: true });
+  },
+};
+```
+
+> The same goes for services
+
+For custom `controllers` and `services` (the ones without a model) you can leave them untouched.
+
+### Migrating routes
+
+In the file `./api/apiName/config/routes.json` we renamed the `destroy` action to `delete` you will have to update your `routes` and `controller` accordingly.
+
+If you haven't customized the `destroy` action then just remove it from your controller instead of renaming it.
+
+**Before**
+
+`./api/product/config/routes.json`
+
+```json
+{
+  "routes": [
+    //...
+    {
+      "method": "DELETE",
+      "path": "/products/:id",
+      "handler": "Product.destroy",
+      "config": {
+        "policies": []
+      }
+    }
+  ]
+}
+```
+
+`./api/product/controllers/Product.js`
+
+```js
+module.exports = {
+  find: async ctx => {
+    return strapi.services.product.fetchAll(ctx.query);
+  },
+
+  findOne: async ctx => {
+    return strapi.services.product.fetch(ctx.params);
+  },
+
+  count: async ctx => {
+    return strapi.services.product.count(ctx.query);
+  },
+
+  create: async ctx => {
+    return strapi.services.product.add(ctx.request.body);
+  },
+
+  update: async ctx => {
+    return strapi.services.product.edit(ctx.params, ctx.request.body);
+  },
+
+  destroy: async ctx => {
+    customImplementation(ctx);
+  },
+};
+```
+
+**After**
+
+`./api/product/config/routes.json`
+
+```json
+{
+  "routes": [
+    //...
+    {
+      "method": "DELETE",
+      "path": "/products/:id",
+      "handler": "Product.delete",
+      "config": {
+        "policies": []
+      }
+    }
+  ]
+}
+```
+
+`./api/product/controllers/Product.js`
+
+```js
+module.exports = {
+  delete: async ctx => {
+    customImplementation(ctx);
+  },
+};
+```
+
+## Migrating `admin`
+
+Customizing the admin is as simple as creating a file in the `./admin` folder of your app. You need to make sure the file you want to customize is at the same location in your `./admin` folder as it is in the `strapi-admin` package. For a reference you can look at the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin).
+
+For the beta there are quite a lot of changes that were made to the admin. If you did customize things you will have to verifiy if the file still exists in the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin) or find its new location.
+
+### Example
+
+You can modify the logo of the app by creating a file `./admin/src/assets/images/logo-strapi.png` to replace the admin logo.
+
+You can do the same with any file in the [source code](https://github.com/strapi/strapi/tree/master/packages/strapi-admin/admin) of the admin panel
+
+## Migrating your database
+
+Starting from `beta` we are adding a new model `Administrator` to replace the `users-permissions` plugin `User` model for connecting to the `Strapi admin panel` panel.
```

Comment:
The `beta` introduces a new Administrator` model that has been created with the single purpose of letting people access the Strapi's administration panel (in the short term this model is not editable0.
In this way, it replaces the previous `User` model from the `users-permissions` plugin. With this new model, you now have a clear distinction between the people that are allowed to access the administration panel, and the users of the application you are building with Strapi.

More practically, it means that you will have a new `strapi_administrator` collection that will be created automatically in your database. On startup, this table will be empty so, when migrating from `alpha` to `beta`, you can either create this new administrator user with the register page OR manually migrate your previous users that have the `administrator` role.

If you face the situation where you have manually migrated your previous admin users you are safe to delete the `Administrator` role (either using the UI or manually in your database) ONLY and ONLY if you don't have any business logic related to the `Administrator` role.
Regarding the previous admin users (the ones that are allowed to access the admin panel) you are safe to delete them if you did not create any relation between the `User` model and another.
Concretely, if you have chosen to migrate your previous admin users in the new `strapi_administrator` collection and your `User` model has a relation with another model you will need to keep both collection in sync. For instance, if you update a user that had the administrator role you will need to apply the same modifications in the other collection.